pkg/apis/config/v1beta1/defaults_test.go

```diff
@@ -0,0 +1,365 @@
+				suggestionConfig.ImagePullPolicy = corev1.PullAlways
+				suggestionConfig.Resources = *newFakeCustomResourceRequirements()
+				return []SuggestionConfig{*suggestionConfig}
+			}(),
+			wantConfig: func() []SuggestionConfig {
+				c := newFakeSuggestionConfig(testAlgorithmName)
+				c.ImagePullPolicy = corev1.PullAlways
+				c.Resources = *newFakeCustomResourceRequirements()
+				return []SuggestionConfig{*c}
+			}(),
+		},
+		fmt.Sprintf("GetSuggestionConfigData sets %s to imagePullPolicy", DefaultImagePullPolicy): {
+			config: func() []SuggestionConfig {
+				suggestion := newFakeSuggestionConfig(testAlgorithmName)
+				suggestion.ImagePullPolicy = ""
+				return []SuggestionConfig{*suggestion}
+			}(),
+			wantConfig: []SuggestionConfig{*newFakeSuggestionConfig(testAlgorithmName)},
+		},
+		"GetSuggestionConfigData sets resource.requests and resource.limits for the suggestion service": {
+			config: func() []SuggestionConfig {
+				suggestion := newFakeSuggestionConfig(testAlgorithmName)
+				suggestion.Resources = corev1.ResourceRequirements{}
+				return []SuggestionConfig{*suggestion}
+			}(),
+			wantConfig: []SuggestionConfig{*newFakeSuggestionConfig(testAlgorithmName)},
+		},
+		fmt.Sprintf("GetSuggestionConfigData sets %s to volumeMountPath", DefaultContainerSuggestionVolumeMountPath): {
+			config: func() []SuggestionConfig {
+				suggestion := newFakeSuggestionConfig(testAlgorithmName)
+				suggestion.VolumeMountPath = ""
+				return []SuggestionConfig{*suggestion}
+			}(),
+			wantConfig: []SuggestionConfig{*newFakeSuggestionConfig(testAlgorithmName)},
+		},
+		"GetSuggestionConfigData sets accessMode and resource.requests for PVC": {
+			config: func() []SuggestionConfig {
+				suggestion := newFakeSuggestionConfig(testAlgorithmName)
+				suggestion.PersistentVolumeClaimSpec = corev1.PersistentVolumeClaimSpec{}
+				return []SuggestionConfig{*suggestion}
+			}(),
+			wantConfig: []SuggestionConfig{*newFakeSuggestionConfig(testAlgorithmName)},
+		},
+		fmt.Sprintf("GetSuggestionConfigData does not set %s to persistentVolumeReclaimPolicy", corev1.PersistentVolumeReclaimDelete): {
+			config: func() []SuggestionConfig {
+				suggestion := newFakeSuggestionConfig(testAlgorithmName)
+				suggestion.PersistentVolumeSpec = corev1.PersistentVolumeSpec{}
+				return []SuggestionConfig{*suggestion}
+			}(),
+			wantConfig: func() []SuggestionConfig {
+				c := newFakeSuggestionConfig(testAlgorithmName)
+				c.PersistentVolumeSpec = corev1.PersistentVolumeSpec{}
+				return []SuggestionConfig{*c}
+			}(),
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			kc := &KatibConfig{
+				RuntimeConfig: RuntimeConfig{
+					SuggestionConfigs: tc.config,
+				},
+			}
+			SetDefaults_KatibConfig(kc)
+			if diff := cmp.Diff(tc.wantConfig, kc.RuntimeConfig.SuggestionConfigs); len(diff) != 0 {
+				t.Errorf("Unexpected SuggestionConfigs (-want,+got):\n%s", diff)
+			}
```

Comment:
Are we going to print the failed test case name here?
Should we try to be consistent for our test-cases, and set the test case name in `testDescription` param to print it?
Like here: https://github.com/kubeflow/katib/blob/42fe2783ed7124f7a3c6717eb902af0a1ef7abec/pkg/controller.v1beta1/experiment/manifest/generator_test.go#L152

Reply:
> Are we going to print the failed test case name here?

Currently implementation, we can print test case name like this:

```shell
go test ./pkg/apis/config/v1beta1/...
--- FAIL: TestSetSuggestionConfigs (0.00s)
    --- FAIL: TestSetSuggestionConfigs/GetSuggestionConfigData_does_not_set_Delete_to_persistentVolumeReclaimPolicy (0.00s)
        defaults_test.go:89: Unexpected SuggestionConfigs (-want,+got):
              []v1beta1.SuggestionConfig{
                {
                        ... // 3 identical fields
                        VolumeMountPath:           "/opt/katib/data",
                        PersistentVolumeClaimSpec: {AccessModes: {"ReadWriteOnce"}, Resources: {Requests: {s"storage": {i: {...}, s: "1Gi", Format: "BinarySI"}}}},
                        PersistentVolumeSpec: v1.PersistentVolumeSpec{
                                ... // 2 identical fields
                                AccessModes:                   nil,
                                ClaimRef:                      nil,
            -                   PersistentVolumeReclaimPolicy: "",
            +                   PersistentVolumeReclaimPolicy: "Delete",
                                StorageClassName:              "",
                                MountOptions:                  nil,
                                ... // 2 identical fields
                        },
                        PersistentVolumeLabels: nil,
                },
              }
FAIL
FAIL    github.com/kubeflow/katib/pkg/apis/config/v1beta1       0.615s
FAIL
```

> Should we try to be consistent for our test-cases, and set the test case name in testDescription param to print it?

I'd like to define test cases as a map and set the description to key. Because the map isn't guaranteed to run orders, and then we can evaluate test cases without depending on the running order.

Reply:
Makes sense, thanks!
@tenzen-y Another good suggestions to add those patterns [in Developer Guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md#developer-guide), so we can keep our unit test consistent across various files.